README.md

```diff
@@ -1 +1,200 @@
-# volttron-emailer
\ No newline at end of file
+volttron-emailer
+=======
+
+The Emailer agent allows an instance of the VOLTTRON platform to send
+email.  When used in combination with the Alert agent, alerts from
+unpublished configured devices will automatically be sent.  In addition,
+agents are able to send emails directly through the pub/sub interface.
+
+Agents needing to send an email through the instance can do so by
+sending the following header and message to the `platform/send_email` topic
+which is monitored by the Emailer agent.  The following 
+is the expected payload for the message body and the optional header.
+
+
+# Prerequisites
+
+* Python 3.8
+
+## Python
+
+<details>
+<summary>To install Python 3.8, we recommend using <a href="https://github.com/pyenv/pyenv"><code>pyenv</code></a>.</summary>
+
+```bash
+# install pyenv
+git clone https://github.com/pyenv/pyenv ~/.pyenv
+
+# setup pyenv (you should also put these three lines in .bashrc or similar)
+export PATH="${HOME}/.pyenv/bin:${PATH}"
+export PYENV_ROOT="${HOME}/.pyenv"
+eval "$(pyenv init -)"
+
+# install Python 3.8
+pyenv install 3.8.10
+
+# make it available globally
+pyenv global system 3.8.10
+```
+</details>
+
+# Installation
+
+1. Create and activate a virtual environment.
+
+    ```shell
+    python -m venv env
+    source env/bin/activate
+    ```
+
+2. Install volttron and start the platform.
+
+    ```shell
+    pip install volttron
+
+    # Start platform with output going to volttron.log
+    volttron -vv -l volttron.log &
+    ```
+
+3. Create you configuration file, in this example the file is named `config`
+
+    ```json
+    {
+        "smtp-address": "<smtp-address>",
+        "smtp-username":"<smtp-username>",
+        "smtp-password":"<smtp-password>",
+        "smtp-port":<smtp-port>,
+        "smtp-tls":<true/false>,
+        "from-address": "foo@foo.com",
+        "to-addresses": [
+            "foo1@foo.com",
+            "foo2@foo.com"
+        ],
+
+        # Only send a certain alert-key message every 120 minutes.
+        "allow-frequency-minutes": 120
+    }
+    ```
+4. Install and start emailer agent.
+
+    ```shell
+    vctl install volttron-emailer --tag email --agent-config <path to config> --start --force
+    ```
+
+## Testing
+
+To verfiy functionality, Get the [send_email.py](./tests/send_email.py) file and run it with
```

Comment:
Perhaps we should tell the user what to expect as output from this and what that script does?  What about a failure what happens?